### PR TITLE
nixos/gdm: switch to rfc42 style settings

### LINKED ISSUE #138627: Add GDM option to hide users list
## Issue description

I'd like to have a NixOS option to make GNOME Display Manager (GDM) hide the list of users that it shows on its log-in screen. This sounds simple enough ([1](https://help.gnome.org/admin/system-admin-guide/stable/login-userlist-disable.html.en), [2](https://www.suse.com/support/kb/doc/?id=000018584)), and I wrote [a patch](https://github.com/8573/nixpkgs/commit/f121f7eccbafb44905e0b77858e465927ffbb173), but I don't understand DConf or know what I'm doing with this GNOME stuff, nor do I know how to test a patch to a NixOS module.

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.62-hardened1, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(root): `"nixos-21.11pre313614.09cd65b33c5"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by Artturin**
TODO add example before merging
###### Things done

as a sidenote i noticed that autologin wasn't working (atleast in a vm) which may be due to the fact that the gdm pam rules are very out of date because they were last updated when they were added many years ago. i'll try updating them

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
@8573 please test https://nixos.wiki/wiki/Nixpkgs/Reviewing_changes#Modules

File: nixos/modules/services/x11/display-managers/gdm.nix [hunk 1]
```diff
@@ -105,6 +107,15 @@ in
         type = types.bool;
       };
 
+      settings = mkOption {
+        type = settingsFormat.type;
+        default = { };
+        description = ''
+          Options passed to the gdm daemon
```
Comment (aanderse) [inline nixos/modules/services/x11/display-managers/gdm.nix:114]: Missing period.

**Review by aanderse — APPROVED** [review]
This is great!

**Comment by Artturin** [comment]
to solve https://github.com/NixOS/nixpkgs/issues/138627
using rf42 settings 
we could use what i've commented her https://github.com/NixOS/nixpkgs/issues/54150#issuecomment-931897717
but its not used anywhere else

**Comment by Artturin** [comment]
Built the options manual and checked that it looks correct

**Pull request merged**